apps/backend/src/app/api/latest/integrations/credential-scanning/revoke/route.tsx

```diff
@@ -0,0 +1,186 @@
+import { getSharedEmailConfig, sendEmail } from "@/lib/emails";
+import { listUserTeamPermissions } from "@/lib/permissions";
+import { getTenancy } from "@/lib/tenancies";
+import { prismaClient } from "@/prisma-client";
+import { createSmartRouteHandler } from "@/route-handlers/smart-route-handler";
+import { KnownErrors } from "@stackframe/stack-shared";
+import { yupNumber, yupObject, yupString } from "@stackframe/stack-shared/dist/schema-fields";
+import { StackAssertionError } from "@stackframe/stack-shared/dist/utils/errors";
+
+export const POST = createSmartRouteHandler({
+  metadata: {
+    summary: "Revoke an API key",
+    description: "Revoke an API key that was found through credential scanning",
+    tags: ["Credential Scanning"],
+  },
+  request: yupObject({
+    body: yupObject({
+      api_key: yupString().defined(),
+    }).defined(),
+  }),
+  response: yupObject({
+    statusCode: yupNumber().oneOf([200]).defined(),
+    bodyType: yupString().oneOf(["success"]).defined(),
+  }),
+  async handler({ body }) {
+
+
+    // Get the API key and revoke it. We use a transaction to ensure we do not send emails multiple times.
+    const updatedApiKey = await prismaClient.$transaction(async (tx) => {
+      // Find the API key in the database
+      const apiKey = await tx.projectApiKey.findUnique({
+        where: {
+          secretApiKey: body.api_key,
+        }
+      });
+
+      if (!apiKey) {
+        throw new KnownErrors.ApiKeyNotFound();
+      }
+
+      if (apiKey.isPublic) {
+        throw new KnownErrors.PublicApiKeyCannotBeRevoked();
+      }
+
+      if (apiKey.manuallyRevokedAt) {
+        return null;
+      }
```

Comment:
can we use the same logic here as the one used to check whether an api key is expired in the other endpoint? eg. you prolly shouldn't be able to revoke an api key that has already expired either